### PR TITLE
create modal for monitoring file copying & fixes for previous solution [FIXED] 

### PR DESCRIPTION
**Pull request by badescuga**
Hi guys,

made the percentage disappear at the same time as the spinner (as requested). I've pushed it to my webapp so you can check it out!

### CODE REVIEW

File: Kudu.Services.Web/Content/Scripts/FileBrowser.js [hunk 1]
```diff
@@ -80,39 +80,51 @@ function showAceHelpModal() {
...
-            //this is used for when copying multiple files in the same time so that i may stii have a coherent percentage
-            copyProgressObjects[uri].transactionPackFinished = false; 
+            this._copyProgressObjects[uri] = {};
+            this._copyProgressObjects[uri].startDate = $.now();
+            this._copyProgressObjects[uri].copyPackEnded = false; //this is used for when copying multiple files in the same time so that i may still have a coherent percentage
         }
 
-        copyProgressObjects[uri].loadedData = loadedData;
-        copyProgressObjects[uri].totalData = totalData;
+        if (totalData == 0) { // empty files appear to have size 0
```
Comment (shrimpy) [inline Kudu.Services.Web/Content/Scripts/FileBrowser.js]: it is better to use "===" than "=="

**Comment by shrimpy** [comment]
@badescuga 

thanks for the fix. just tried, mostly looks good. but notice one issue. If open modal dialog after reaching 100%. the Modal is empty.

see below two video for detail repro

http://1drv.ms/1G64fCH
http://1drv.ms/1G64ltO
(PS: Video in second link, i did two hard refresh ctrl+f5 at the beginning)

**Comment by badescuga** [comment]
Am able to reproduce this. This is very probably caused by the lack of
updating the observable object. This didn't manifest itself till now
because the files that were transfered updated the observable object. At
100%, there is no one updating the observable.

Sorry for not debugging accordingly. Will have a fix shortly.

**Comment by badescuga** [comment]
fixed the 100% issue. You can also check it out on my dev site.

File: Kudu.Services.Web/Kudu.Services.Web.csproj [hunk 2]
```diff
@@ -15,6 +15,7 @@
     <IISExpressAnonymousAuthentication />
     <IISExpressWindowsAuthentication />
     <IISExpressUseClassicPipelineMode />
+    <UseGlobalApplicationHostFile />
```
Comment (shrimpy) [inline Kudu.Services.Web/Kudu.Services.Web.csproj]: unnecessary change?

**Comment by shrimpy** [comment]
@badescuga looks good. i will see if other ppl have any comments. if not i will integrate your changes. Thanks again for doing this.

**Comment by badescuga** [comment]
My pleasure! Will start working on other stuff for the portal. I promise i'll be more careful with the ux, will stop testing it at 1 a.m. in the morning :).


 If you'd like any specific feature, just let me know. I'm sure i can be of help!



On Thu, Oct 15, 2015 at 11:26 AM -0700, "Xiaomin Wu" <notifications@github.com> wrote:












@badescuga looks good. i will see if other ppl have any comments. if not i will integrate your changes. Thanks again for doing this.

**Comment by shrimpy** [comment]
@badescuga 

once you fixed my last comment. can you also do a "rebase -i" against our master branch, to combine all your changes into one?

i tried to integrate you PR, but there are some conflicts, i think will be good to let you resolve then.

once you did that i will integrate your PR into our master branch.

File: Kudu.Services.Web/DebugConsole/Default.cshtml [hunk 3]
```diff
@@ -1,147 +1,187 @@
-﻿@{
```
Comment (davidebbo) [inline Kudu.Services.Web/DebugConsole/Default.cshtml]: Do you know what's causing this file do show up as being 100% different here, despite it being mostly the same? Maybe there is a line ending issue with your editor?

I wonder if the fact that we're missing *.cshtml from the .gitattributes contributes to the problem.

**Comment by davidebbo** [comment]
I rebased it and merged it.
